### PR TITLE
fix(lookup/grafana_dashboard): add custom certs verification logic

### LINKED ISSUE #346: grafana_dashboard lookup plugin does not accept `validate_certs` as a parameter and does not perform accordingly
##### SUMMARY

As a simple example, I have a Grafana Container running behind a reverse-proxy with Self-Signed Certificates which I do not want to have validated. The Grafana instance is reachable via `https://<remote_machine>/grafana` API

Upon using the following task:

```yaml
- ansible.builtin.set_fact:
    current_dashboards: |
      {{
        lookup(
          'community.grafana.grafana_dashboard',
          'grafana_url=https://{{ inventory_hostname }}/grafana grafana_user=admin grafana_password=test search=foo',
          validate_certs=false
        )
      }}
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

`community.grafana.grafana_dashboard` lookup [plugin](https://docs.ansible.com/ansible/latest/collections/community/grafana/grafana_dashboard_lookup.html#ansible-collections-community-grafana-grafana-dashboard-lookup)

##### ANSIBLE VERSION

```paste below
ansible [core 2.13.8]
  config file = /mnt/c/Users/E1401748/Development/gitlab.emrsn.org/pacedge-3.x/appstack/ansible.cfg
  configured module search path = ['/home/shantanoo/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/shantanoo/.local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/shantanoo/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.8.10 (default, May 26 2023, 14:05:08) [GCC 9.4.0]
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```paste below
# /home/shantanoo/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.grafana 1.7.0  
```

##### CONFIGURATION

```paste below
DEFAULT_MANAGED_STR(/mnt/c/Users/E1401748/Development/gitlab.emrsn.org/pacedge-3.x/appstack/ansible.cfg) = This file is generated by Manager.%n
Generated On: %Y-%m-%d %H:%M:%S %n
DEFAULT_STDOUT_CALLBACK(/mnt/c/Users/E1401748/Development/gitlab.emrsn.org/pacedge-3.x/appstack/ansible.cfg) = yaml
```

##### OS / ENVIRONMENT

Windows Subsystem For Linux 2 - Ubuntu 20.04 LTS

##### STEPS TO REPRODUCE

```yaml
- hosts: all
  tasks:
    - ansible.builtin.set_fact:
        dashboards: "{{ lookup('community.grafana.grafana_dashboard', 'grafana_url=https://{{ inventory_hostname}}/grafana grafana_user=admin grafana_password=test search=foo', validate_certs=false) }}"
    - debug: var=dashboards
```

##### EXPECTED RESULTS

Additional `validate_certs` variable should be accepted to bypass SSL certificate validation, the value is either `true` or `false`

An expectation is that the lookup plugin might behave similarly to the URL lookup plugin where explicit headers and `validate_certs` parameters can be explicitly set.
##### ACTUAL RESULTS

Throws a SSL certificate verification error:

```
An unhandled exception occurred while running the lookup plugin ''community.grafana.grafana_dashboard''. Error was a <class ''urllib.error.URLError''>, original message: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1131)>. <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1131)>'
```

### PR DESCRIPTION
**Pull request by shantanoo-desai**
##### SUMMARY

This fix adds the `validate_certs`, `ca_path` options to the lookup plugin. Both parameters comply with the `get_url` functionality of ansible-core and provides additional utility to perform lookup of dashboards from Grafana instances that are configured with Self-Signed Certificates.

`validate_certs` option value defaults to `true` - following the pattern of `url` lookup plugin from the Core.

`ca_path` option value is set explicitly when using the plugin else defaults to `None`.

Fixes #346

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

Lookup Plugin for Grafana Dashboards

##### ADDITIONAL INFORMATION

Tested with a Grafana Docker image with Self-Signed Certificates as well as a local Certificate Authority. The Local CA along with a self-signed certificate is used to form a `chain.crt` and validated as `ca_path` option for the plugin.

### CODE REVIEW

**Comment by shantanoo-desai** [comment]
@Nemental One of the CI Sanity check for development fails for a test file which doesn't belong to my PR. The other Lint problem will be solved by me now. I am not sure why the particular test file for Sanity Develop Job fails?

**Comment by Nemental** [comment]
Hey @shantanoo-desai 
Thanks for your PR!
I'll do a final review of the changes tomorrow, but so far it looks pretty good. The remaining sanity failings have already been fixed in another PR.

**Comment by shantanoo-desai** [comment]
## Update

the two Integration Tests seem to fail with __HTTP 429 Too Many Requests__ on the Grafana Endpoint when trying to retrieve some dashboards. I am assuming the tests would just need to be re-run and nothing should actually from patch should break the tests.

**Comment by shantanoo-desai** [comment]
## Update

the two Integration Tests seem to fail with __HTTP 429 Too Many Requests__ on the Grafana Endpoint when trying to retrieve some dashboards. I am assuming the tests would just need to be re-run and nothing should actually from patch should break the tests.

File: plugins/lookup/grafana_dashboard.py [hunk 1]
```diff
@@ -51,13 +58,25 @@
...
     grafana_dashboards: "{{ lookup('grafana_dashboard', 'grafana_url=http://grafana.company.com grafana_api_key=' ~ grafana_api_key) }}"
+
+- name: get project foo grafana dashboards (validate SSL certificates of the instance with custom CA Certificate Bundle)
+  set_fact:
+    grafana_dashboards: |
+        {{
+            lookup(
+              'grafana_dashboard',
+              'grafana_url=https://grafana.company.com grafana_user=admin grafana_password=admin search=foo',
+              validate_cert=true,
```
Comment (rndmh3ro) [inline plugins/lookup/grafana_dashboard.py:69]:
```suggestion
              validate_certs=true,
```

File: plugins/lookup/grafana_dashboard.py [hunk 2]
```diff
@@ -41,6 +41,13 @@
     description: optional filter for dashboard search.
     env:
       - name: GRAFANA_DASHBOARD_SEARCH
+  validate_certs:
+    description: flag to control SSL certificate validation
+    type: boolean
+    default: True
```
Comment (Nemental) [inline plugins/lookup/grafana_dashboard.py:47]:
```suggestion
    default: true
```

**Comment by shantanoo-desai** [comment]
@Nemental and @rndmh3ro The PR is rebased to main branch and the changes have been made. The Sanity tests still seem to fail however.

**Comment by Nemental** [comment]
The remaining sanity workflow failures are not related with your changes and have already been fixed in #354 :)

**Review by Nemental — APPROVED** [review]
lgtm! 🥳

**Pull request merged**